### DISCUSSION TITLE
Azure Blob File

### DISCUSSION THREAD

**Discussion by ykim22153** [OvertureMaps/data#70] [Discussion]
I've been trying to interface with the Azure blob storage endpoint in my python script.  S3 works fine however when I use the same pattern URL as with AWS I get an error that there is no file.  my read line is below:

read_parquet('https://overturemapswestus2.blob.core.windows.net/release/2023-07-26-alpha.0/theme=admins/type=*/*', hive_partitioning=1)

Am I call the correct endpoint using DuckDB?

**Comment by varapmsft** [comment]
Looks like DuckDB CLI does not support Azure blob storage endpoint yet. I tried to use the 'Azure' extension from their documentation but I was not able to install it.
https://duckdb.org/docs/extensions/azure.

Another option perhaps is to use Python libraries, like mentioned in the below page.
https://datamonkeysite.com/2023/01/27/querying-azure-storage-using-duckdb/

**Comment by varapmsft** [comment]
@ykim22153 I followed up with duckdb folks and they fixed the issue in the latest version 0.9.1. I am able to query Azure location now. One issue still pending is that the  columns in the folder paths (theme and type) are not yet queryable.

INSTALL azure;
LOAD azure;
INSTALL spatial;
LOAD spatial;
SET azure_storage_connection_string = 'DefaultEndpointsProtocol=https;AccountName=overturemapswestus2;AccountKey=;EndpointSuffix=core.windows.net';
	   
SELECT
		   subType,
           localityType,
           adminLevel,
           isoCountryCodeAlpha2,
           JSON(names) AS names,
           JSON(sources) AS sources,
           ST_GeomFromWkb(geometry) AS geometry
      FROM read_parquet('azure://release/2023-07-26-alpha.0/theme=admins/type=*/*')
     WHERE adminLevel = 2
       AND ST_GeometryType(ST_GeomFromWkb(geometry)) IN ('POLYGON','MULTIPOLYGON')
	   LIMIT 10;
